src/store/messages/saga.ts

```diff
@@ -486,6 +486,12 @@ export function* replaceOptimisticMessage(currentMessages, message) {
 export function* receiveUpdateMessage(action) {
   let { message } = action.payload;
 
+  // Check if the message already exists in the state
+  const existingMessage = yield select(messageSelector(message.id));
+  if (existingMessage && existingMessage.message === message.message) {
+    return;
+  }
```

Comment:
I'm not sure this is the right solution. A message could be updated for any number of reasons that aren't the message changing.

We should be able to update this message in state and if nothing has changed then it should just work. If that's not working then we're updating too much of the state or we're not providing enough of it in the event.

Reply:
Hm yeah looks like that shouldn't be needed. 

The reason for the edited message rendering as a 'received' message is due to the `userId` and `senderId` differing slightly, therefore the `isOwner` boolean is false when determining if sent/edited message is from the current user. Maybe we could have a v.quick call to discuss?

<img width="872" alt="Screenshot 2023-10-18 at 11 41 46" src="https://github.com/zer0-os/zOS/assets/39112648/3d863f60-6fea-4fdc-a87c-b6b3f3214510">

Reply:
i guess this is related to NOT mapping the user correctly maybe, which receiving the edit message event perhaps? (i.e userId/senderId don't match because one of them is zOS user id, and other one is matrixId). Could you please check/confirm?